File: Assets/EPICGODDAMNWORLDGENDEMO/PathGen/Utils/ChunkHelperFunctions.cs
using UnityEngine;

public static class ChunkHelperFunctions
{
    static int gridSize = 3;

    public static void SetGridSize(int g)
    {
        gridSize = g;
    }

    public static readonly (Vector3Int dir, DirectionEnum directionEnum)[] directions =
    {
        (Vector3Int.up, DirectionEnum.Up),
        (Vector3Int.down, DirectionEnum.Down),
        (Vector3Int.left, DirectionEnum.Left),
        (Vector3Int.right, DirectionEnum.Right),
    };

    public static int ParseLocationToId(Vector3Int loc)
    {
        return loc.x * gridSize * gridSize
             + loc.y * gridSize
             + loc.z;
    }

    public static bool IsInsideGrid(Vector3Int pos)
    {
        return pos.x >= 0 && pos.x < gridSize &&
               pos.y >= 0 && pos.y < gridSize &&
               pos.z >= 0 && pos.z < gridSize;
    }

    public static Vector3Int DirectionToVector(DirectionEnum dir)
    {
        return dir switch
        {
            DirectionEnum.Up => Vector3Int.up,
            DirectionEnum.Down => Vector3Int.down,
            DirectionEnum.Left => Vector3Int.left,
            DirectionEnum.Right => Vector3Int.right,
            _ => Vector3Int.zero
        };
    }

    public static DirectionEnum GetOpposite(DirectionEnum dir)
    {
        return dir switch
        {
            DirectionEnum.Up => DirectionEnum.Down,
            DirectionEnum.Down => DirectionEnum.Up,
            DirectionEnum.Left => DirectionEnum.Right,
            DirectionEnum.Right => DirectionEnum.Left,
            _ => dir
        };
    }


}

File: JavaMinecraftProject/untitled/src/main/java/com/daan/spelunky3D/pathGen/utils/ChunkHelperFunctions.java
package com.daan.spelunky3D.pathgen.utils;

import com.daan.spelunky3D.pathgen.enums.DirectionEnum;
import com.daan.spelunky3D.pathgen.models.Vector3Int;

public class ChunkHelperFunctions {

    private static int gridSize = 3;

    public static void setGridSize(int g) {
        gridSize = g;
    }

    public static final DirectionEntry[] directions = {
            new DirectionEntry(new Vector3Int(0, 1, 0), DirectionEnum.NORTH),
            new DirectionEntry(new Vector3Int(0, -1, 0), DirectionEnum.SOUTH),
            new DirectionEntry(new Vector3Int(-1, 0, 0), DirectionEnum.WEST),
            new DirectionEntry(new Vector3Int(1, 0, 0), DirectionEnum.EAST)
    };

    public static int parseLocationToId(Vector3Int loc) {
        return loc.x * gridSize * gridSize
                + loc.y * gridSize
                + loc.z;
    }

    public static boolean isInsideGrid(Vector3Int pos) {
        return pos.x >= 0 && pos.x < gridSize &&
                pos.y >= 0 && pos.y < gridSize &&
                pos.z >= 0 && pos.z < gridSize;
    }

    public static Vector3Int directionToVector(DirectionEnum dir) {
        return switch (dir) {
            case NORTH -> new Vector3Int(0, 1, 0);
            case SOUTH -> new Vector3Int(0, -1, 0);
            case WEST -> new Vector3Int(-1, 0, 0);
            case EAST -> new Vector3Int(1, 0, 0);
            default -> new Vector3Int(0, 0, 0);
        };
    }

    public static DirectionEnum getOpposite(DirectionEnum dir) {
        return switch (dir) {
            case NORTH -> DirectionEnum.SOUTH;
            case SOUTH -> DirectionEnum.NORTH;
            case WEST -> DirectionEnum.EAST;
            case EAST -> DirectionEnum.WEST;
            default -> dir;
        };
    }
}